### PR TITLE
feat(multiselect): add multiSelect weak emphasis

### PR DESCRIPTION
**Pull request by joseazevedo88**
New variant for asset audit trail in `MultiSelect` component. This component should still work retroactively

image

### CODE REVIEW

File: src/components/Menu/Menu.tsx
Comment (fabriciomendonca) [inline src/components/Menu/Menu.tsx:21]: Just saw a problem with the `Menu` z-index I let pass during the development. You can see on the component Docs page on Storybook by opening the Basic Menu With Links. It will be behind the Menu Without Trigger section.
Adding the class `tw-z-[120000]` to the base classes will solve the issue (this is the same class applied on the `Tooltip` component)

File: src/components/Menu/Menu.cy.tsx
Comment (fabriciomendonca) [inline src/components/Menu/Menu.cy.tsx:64]: This block was supposed to add focus to the menu without a trigger. I saw now it is auto focusing the first element by default because of the keyboard navigation hook.
I was thinking here: maybe this can be an issue if in a page that uses the `Menu` without a trigger, the keyboard navigation will start on that menu first item. 
WDYT about changing the keyboard navigation hook to not autofocus it if no trigger element?
cc: @imoutaharik

File: src/components/Trigger/Trigger.tsx
Comment (fabriciomendonca) [inline src/components/Trigger/Trigger.tsx:17]: This will add a breaking change, if it is being used outside of Fondue.
I checked Clarify and couldn't find any usage of it.
Also checked that it was created by @Walkeezy [here](https://github.com/Frontify/fondue/pull/1076/files).
Idk if he is using outside of Fondue, so it would be nice to double check if these changes will not break anything before merging them.

Comment (Walkeezy) [reply src/components/Trigger/Trigger.tsx:17]: Going to use it in Frontify/guideline-blocks. It's fine to rename it from my side. :-)

Comment (joseazevedo88) [reply src/components/Trigger/Trigger.tsx:17]: Nice, thanks. Let's do `TriggerEmphasis` for consistency then 👍

**Comment by thisguyscode** [comment]
@joseazevedo88 There are quite a few interaction issues that I think need addressing before we release. I don't think that a comment is going to do it really. Happy to do a loom but best would be a meeting.

File: src/components/MultiSelect/MultiSelect.tsx
Comment (fabriciomendonca) [inline src/components/MultiSelect/MultiSelect.tsx:16]: maybe we should use from `@components/MenuItem` for consistency?

File: src/components/MultiSelect/MultiSelect.tsx
Comment (fabriciomendonca) [inline src/components/MultiSelect/MultiSelect.tsx:156]: you can remove the `else` statement here and just use `return ''` since it will return inside the `if`

File: src/components/MultiSelect/MultiSelect.tsx
Comment (fabriciomendonca) [inline src/components/MultiSelect/MultiSelect.tsx:170]: I believe you can simplify this logic, WDYT?

```suggestion
        if (emphasis === MultiSelectEmphasis.Weak || !open) {
            return TagType.Selected;
        }

        return TagType.SelectedWithFocus;
```

File: src/components/MultiSelect/MultiSelect.tsx
Comment (fabriciomendonca) [inline src/components/MultiSelect/MultiSelect.tsx:107]: The `ESC` key when focusing on the menu was supposed to close the menu but it is not working here. You can see it working on the `Menu` component stories

**Review by fabriciomendonca — CHANGES_REQUESTED** [review]

**Comment by joseazevedo88** [comment]
@thisguyscode this is not helpful. First, I don't know who you are. Second, I'm not part of fondue team and my task is to quickly prototype something so we can release and then fondue team can iterate over it and accomplish the original design vision. Therefore, this will need to be merged fast so we can get rid of fondue dependencies on my(analytics) side. Third, if there are broken interactions with the previous implementation kindly let me know via comment, or loom, so I can address them. Thanks

**Comment by thisguyscode** [comment]
@joseazevedo88 Hey, it's Jamie :D

Didn't mean to offend you or anything, just though it might be easier to jump on a call, and if you were not available for a call then I could record a loom.

I'll record a loom.

**Comment by joseazevedo88** [comment]
Hey @thisguyscode Jamie 👋 😄  No offense taken, I was just wondering if who commented had the whole context 😅 Please also sync with Nikolai on it, having the input inside a button is an a11y violation(and just code in general). Also in this case(see 1st screenshot) how can the user know there's an input there? That's why I added the possibility to pass a label to that input(see 2nd screenshot). I'm up for a meeting to discuss this if needed.
image

image

File: src/components/MultiSelect/MultiSelect.tsx
Comment (fabriciomendonca) [reply src/components/MultiSelect/MultiSelect.tsx:156]: It is not about saving two lines of code. It is about avoiding future unreachable code if any written after the `else`. And It would be ok if you decided not to change it now. I know there is more than one solution for a problem, and it is great when we use this space for at least pointing them and learning from each other. I'm sorry if it looked I was only trying to save two lines of code.

**Comment by thisguyscode** [comment]
Bugs/regressions:
https://www.loom.com/share/c5eb1dcf6b144adebb37f8d57a13dc95
https://www.loom.com/share/9416ded66ae9404f9e138e9bcdae38ca
https://www.loom.com/share/a2f3b5a2a5d642c2b5c0497be89019b7

UX suggestion:
https://www.loom.com/share/e07847f51d404dbe82d69515940d6486

There may be more but those ones make it hard to fully assess from a usability standpoint.

**Review by fabriciomendonca — APPROVED** [review]
Approved after a peer review with @joseazevedo88

**Pull request merged**
